From: Thomas Renninger <trenn@suse.de>
Subject: intel_idle: Fix kvm -cpu host Null pointer enter function issue
Reference: bnc#726296
Patch-Mainline: not yet

kvm -cpu host paramter exports the native cpuid, but not all features
intel_idle expects...

Signed-off-by: Thomas Renninger <trenn@suse.de>

---
 drivers/idle/intel_idle.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

Index: linux-3.1-12.1/drivers/idle/intel_idle.c
===================================================================
--- linux-3.1-12.1.orig/drivers/idle/intel_idle.c
+++ linux-3.1-12.1/drivers/idle/intel_idle.c
@@ -321,9 +321,12 @@ static int intel_idle_probe(void)
 
 	cpuid(CPUID_MWAIT_LEAF, &eax, &ebx, &ecx, &mwait_substates);
 
+	/* mwait substates can be zero in kvm case with -cpu host
+	   exporting native cpuid, but not all cpuid features */
 	if (!(ecx & CPUID5_ECX_EXTENSIONS_SUPPORTED) ||
-		!(ecx & CPUID5_ECX_INTERRUPT_BREAK))
-			return -ENODEV;
+	    !(ecx & CPUID5_ECX_INTERRUPT_BREAK) ||
+	    !mwait_substates)
+		return -ENODEV;
 
 	pr_debug(PREFIX "MWAIT substates: 0x%x\n", mwait_substates);
 
